### DISCUSSION TITLE
How do I set up TLS for a private instance?

### DISCUSSION THREAD

**Discussion by vext01** [astubenbord/paperless-mobile#404] [Q&A] [answered]
Hi,

Seems the android app won't connect to my instance, presumably because TLS isn't set up: it's a private instance that I don't want exposed to the internet.

There's an option to set up a mutual tls cert, but I have no idea how to do this.

What are others in the same situation doing?

**Comment by florianschroen** [comment, marked as answer]
You can try using letsencrypt with dns challenge which only requires a compatible dns provider for your Domain. (which needs to be an normal registered Domain)

When you like to use your service only via local network without exposing to the Internet, you can create a dns record of the type A with your private Server IP.

E.g. paperless.mydomain.tld will point to 192.168.178.10

This setup is technically absolutely acceptible and secure.

I would recommend looking at traefik loadbalancer, because it can do all the certificate stuff very easily.

**Reply by vext01** [reply]
Thanks. After a bit of research I came to the same answer. A bit more involved than I had hoped, but such is life :)
